File: code-samples/voice/code-snippets/call-monitoring-group.py
from ringcentral import SDK
import json

#
# Read all call monitoring groups
#
def read_call_monitoring_groups():
    try:
        endpoint = "/restapi/v1.0/account/~/call-monitoring-groups"
        resp = platform.get(endpoint)
        jsonObj = resp.json()
        for group in jsonObj.records:
            print (f'Call monitoring group name: {group.name} / {group.id}')
            read_call_monitoring_group_members(group.id)
    except Exception as e:
      print ("Unable to call list call monitoring groups." + str(e))

#
# Read a call monitoring group members
#
def read_call_monitoring_group_members(groupId):
    try:
        endpoint = f'/restapi/v1.0/account/~/call-monitoring-groups/{groupId}/members'
        resp = platform.get(endpoint)
        jsonObj = resp.json_dict()
        print ("Call monitoring group members:")
        for member in jsonObj['records']:
            print (json.dumps(member, indent=2, sort_keys=True))
    except Exception as e:
        print ("Unable to read members of this call monitoring group." + str(e))

# Authenticate a user using a personal JWT token
def login():
  try:
      platform.login( jwt= "SANDBOX-JWT" )
      read_call_monitoring_groups()
  except Exception as e:
      print ("Unable to authenticate to platform. Check credentials. " + str(e))

File: code-samples/voice/code-snippets/call-monitoring-group.rb
require 'ringcentral'
require 'json'

#
# Read all call monitoring groups
#
def read_call_monitoring_groups()
    begin
        endpoint = "/restapi/v1.0/account/~/call-monitoring-groups"
        resp = $platform.get(endpoint)
        jsonObj = resp.body
        for group in jsonObj['records'] do
            puts ("Call monitoring group name: " + group['name'] + "/"+ group['id'])
            read_call_monitoring_group_members(group['id'])
        end
    rescue StandardError => e
      puts ("Unable to call list call monitoring groups." + e.to_s)
    end
end
#
# Read a call monitoring group members
#
def read_call_monitoring_group_members(groupId)
    begin
        endpoint = "/restapi/v1.0/account/~/call-monitoring-groups/" + groupId + "/members"
        resp = $platform.get(endpoint)
        jsonObj = resp.body
        puts ("Call monitoring group members:")
        for member in jsonObj['records'] do
            puts JSON.pretty_generate(member)
        end
    rescue StandardError => e
        puts ("Unable to read members of this call monitoring group." + e.to_s)
    end
end


# Authenticate a user using a personal JWT token
def login()
  begin
    $platform.authorize( jwt: "RC_USER_JWT" )
    read_call_monitoring_groups()
  rescue StandardError => e
    puts ("Unable to authenticate to platform. Check credentials. " + e.to_s)
  end
end
